addons/mail/static/src/messaging/component/message_list/message_list.js

```diff
@@ -166,23 +166,12 @@ class MessageList extends Component {
     /**
      * @returns {mail.messaging.component.Message|undefined}
      */
-    get lastCurrentPartnerMessageRef() {
-        const currentPartnerMessageRefs = this.messageRefs.filter(messageRef =>
-            (
-                messageRef.message.author &&
-                messageRef.message.author === this.env.messaging.currentPartner
-            )
-        );
-        const { length: l, [l - 1]: lastCurrentPartnerMessageRefs } = currentPartnerMessageRefs;
-        return lastCurrentPartnerMessageRefs;
-    }
-
-    /**
-     * @returns {mail.messaging.component.Message|undefined}
-     */
-    get lastMessageRef() {
-        const { length: l, [l - 1]: lastMessageRef } = this.messageRefs;
-        return lastMessageRef;
+    get mostRecentMessageRef() {
+        if (this.props.order === 'desc') {
+            return this.messageRefs[0];
+        }
```

Comment:
The method should probably preemptively return if `this.messageRefs` is falsy or has length 0.

Reply:
`messageRefs` is always an array, and `[0]` on empty array always return `undefined`, which is valid per docstring

Reply:
I feel like this is relying too much on an implementation detail, but fine either way.